### DISCUSSION TITLE
tiggering a mission on extance of a ship in the fleet

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#7832] [Q&A] [answered]
Greetings,

is there a way to trigger a mission when a specific ship is part of  a fleet?

**Comment by mOctave** [comment, marked as answer]
Yes. There is. See https://github.com/endless-sky/endless-sky/wiki/Player-Conditions#Read-only.

**Reply by Zitchas** [reply]
I don't think we can, actually. We can check for specific *models* or *categories* of ships, but not a specific individual ship. (although we do have UID, so the framework is getting there for doing so)

**Reply by mOctave** [reply]
I suppose it depends on how you parse the question. I took it to be asking if it was possible to check for a specific model.

**Reply by Zitchas** [reply]
Fair, yes. And we generally talk about models, not individual ships. Particularly since individual ships are almost never referred to (other than the Emerald Sword, Archons, and Author ships.)

**Reply by daggs1** [reply]
> Yes. There is. See https://github.com/endless-sky/endless-sky/wiki/Player-Conditions#Read-only.

it seems that it is only for npc, am I correct?

**Reply by Zitchas** [reply]
> > Yes. There is. See https://github.com/endless-sky/endless-sky/wiki/Player-Conditions#Read-only.
> 
> it seems that it is only for npc, am I correct?

I think some clarification is required: Are you looking to check that a specific, individual, ship is part of the player's fleet? Or are you looking to see if the player has a particular *model* of ship in their fleet?

For example, do you want to check if the player has *any* starbarges in their fleet? Or do you want to check if they have a specific starbarge that was given to them in a previous mission in their fleet?

edit: "Starbarge in general" vs "This one specific individual Starbarge"

**Reply by daggs1** [reply]
I want to add another mission related to the Emerald Sword. it should depend on the ship reaching a system by either player's flagship or npc ship

**Reply by mOctave** [reply]
So if you mean that you want to add a mission that only triggers if you have the Emerald Sword, then yes. The Emerald Sword is one-of-a-kind and so you can address it based off of the model name. Conditions would be:

`"ship model: Emerald Sword" > 0` (`"ship model: Emerald Sword"` might work too) if you want it to work if you have the Emerald Sword in your active fleet, or
`"flagship model: Emerald Sword"` if you want it to work if you have the Emerald Sword as your flagship but not otherwise.

**Reply by daggs1** [reply]
thanks, I'll try that
